### PR TITLE
Review of the Silverstripe CMS 5.2.0 changelog

### PR DESCRIPTION
**Pull request by GuySartorelli**
## Description
Review of the Silverstripe CMS 5.2.0 changelog.

### Commits
1. Update note about linkfield migration guides
2. add new items to changelog
3. overall review of content (writing style and readability)

### New items added to the changelog

I found these in the commits list from the beta changelog. If you disagree with any of these being in the changelog let me know, there's only a couple here that I'm really adamant on and the rest I could take or leave.

- https://github.com/silverstripe/silverstripe-framework/pull/11119
- https://github.com/silverstripe/silverstripe-admin/pull/1678
- https://github.com/silverstripe/silverstripe-admin/pull/1658 maybe?
- https://github.com/silverstripe/silverstripe-staticpublishqueue/pull/178
- deprecations in https://github.com/silverstripe/silverstripe-framework/pull/11049

## Issues

- https://github.com/silverstripe/developer-docs/issues/472

## Pull request checklist

- [x] The target branch is correct
    - See [branches and commit messages](https://docs.silverstripe.org/en/contributing/documentation#branches-and-commit-messages)
- [x] All commits are relevant to the purpose of the PR (e.g. no TODO comments, unrelated rewording/restructuring, or arbitrary changes)
    - Small amounts of additional changes are usually okay, but if it makes it hard to concentrate on the relevant changes, ask for the unrelated changes to be reverted, and submitted as a separate PR.
- [x] The commit messages follow our [commit message guidelines](https://docs.silverstripe.org/en/contributing/code/#commit-messages)
- [x] The PR follows our [contribution guidelines](https://docs.silverstripe.org/en/contributing/documentation/)
- [x] The changes follow our [writing style guide](https://docs.silverstripe.org/en/contributing/documentation/#writing-style)
- [x] Code examples follow our [coding conventions](https://docs.silverstripe.org/en/contributing/coding_conventions/)
- [ ] CI is green

### CODE REVIEW

File: en/02_Developer_Guides/00_Model/08_SQL_Select.md [hunk 1]
```diff
@@ -418,10 +418,10 @@ Common Table Expressions are a powerful tool both for optimising complex queries
...
-if (DB::get_conn()->supportsCteQueries()) {
-    // Supports non-recursive CTE clause
-} elseif (DB::get_conn()->supportsCteQueries(true)) {
+if (DB::get_conn()->supportsCteQueries(true)) {
     // Supports recursive CTE clause
+} elseif (DB::get_conn()->supportsCteQueries()) {
+    // Supports non-recursive CTE clause
 } else {
     // No CTE support
 }
```
Comment (ChloeSartorelli) [inline en/02_Developer_Guides/00_Model/08_SQL_Select.md:427]: Unrelated change - with the old code, the `elseif` condition would never have been executed because if it doesn't support *non*-recursive queries it definitely doesn't support recursive ones.

File: en/02_Developer_Guides/00_Model/09_Validation.md
Comment (ChloeSartorelli) [inline en/02_Developer_Guides/00_Model/09_Validation.md:1]: Changes here reflect the changes in the changelog for the same content

File: en/10_Contributing/02_Documentation.md
Comment (ChloeSartorelli) [inline en/10_Contributing/02_Documentation.md:1]: Unrelated changes - just noticed a couple of cases where this list about the writing style doesn't follow its own rules.

File: en/10_Contributing/02_Documentation.md [hunk 2]
```diff
@@ -94,16 +94,15 @@ yarn lint --fix
...
-- If referring to maintainers of Silverstripe CMS, either explicitly say "the maintainers", or write in second person plural form (use "we" instead of "I")
+- If referring to maintainers of Silverstripe CMS, either explicitly say "the maintainers", or write in second person plural form (use "we" instead of "I").
 - Use gender neutral language throughout the document, unless referencing a specific person. Use them, they, their, instead of he and she, his or her.
 - Use simple language and words. Avoid uncommon jargon and overly long words. Remember that not everyone speaks English as their primary language.
 - Use UK English, not US English. Silverstripe CMS is proudly a New Zealand open source project we use the UK spelling and forms of English. The most common of these differences are -ize vs -ise, or -or vs our (eg color vs colour).
 - Write in an active and direct voice.
 - Avoid saying words like "obviously" or "of course". Things that are obvious to you as the writer may not be so obvious to the person reading the documentation.
-- Keep documentation lines shorter than 120 characters.
```
Comment (ChloeSartorelli) [inline en/10_Contributing/02_Documentation.md:103]: We decided to explicitly not enforce this, so we shouldn't say it's a rule

File: en/_images/changelogs/5.2.0/elemental-badge.png
Comment (ChloeSartorelli) [inline en/_images/changelogs/5.2.0/elemental-badge.png:1]: Should be a little less blurry since it's larger, and shows both of the versioned labels

File: en/08_Changelogs/5.2.0.md [hunk 3]
```diff
@@ -29,7 +29,9 @@ title: 5.2.0 (unreleased)
 
 ## Security considerations {#security-considerations}
 
-This release includes several security fixes. Review the individual vulnerability disclosure for more detailed descriptions of each security fix. We highly encourage upgrading your project to include the latest security patches.
+Some security fixes that were previously released in the January security release that are mentioned in the [Silverstripe CMS security patches January 2024](https://www.silverstripe.org/blog/silverstripe-cms-security-patches-january-2024/) blog post are listed below.
+
+Review the individual vulnerability disclosure for more detailed descriptions of each security fix. We highly encourage upgrading your project to include the latest security patches.
```
Comment (ChloeSartorelli) [inline en/08_Changelogs/5.2.0.md:34]: Copied over from the beta changelog

**Review by sabina-talipova — APPROVED** [review]
LGTM
